### PR TITLE
Enable tbb use in embree3

### PR DESCRIPTION
**Pull request by AlvaroFS**
Allow compiling **embree3** with **tbb/onetbb** as task system
---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AlvaroFS** [comment]
I'm not sure whether the error comes from using  `self.info.settings` instead of `self.settings` since the check inside the `validate` method seems fine. The linter reports there is no `info` field in `self`. I already reported this in the webinar and @uilianries responded, but I should specify that I wasn't using a header only lib. This is one example.

File: recipes/embree3/all/conanfile.py [hunk 1]
```diff
@@ -60,8 +66,11 @@ class EmbreeConan(ConanFile):
         "ray_masking": False,
         "backface_culling": False,
         "ignore_invalid_rays": False,
+        "with_tbb": False,
     }
 
+    generators = "CMakeDeps"
+
```
Comment (uilianries) [inline recipes/embree3/all/conanfile.py:73]:
```suggestion
```

Please, avoiding mixing generators here, it makes harder to review. Move it to generate()

**Comment by uilianries** [comment]
@AlvaroFS please, keep self.info on validate(). This recipe is not header-only.

**Review by ericLemanissier — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by prince-chrismc** [comment]
An agent hanged 😖

**Pull request merged**